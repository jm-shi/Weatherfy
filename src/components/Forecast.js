import React from 'react';
import Navbar from './Navbar';
import Today from './Today';
import FiveDay from './FiveDay';
import PropTypes from 'prop-types';

class Forecast extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showTodayWeather: true
    };
    this.setToday = this.setToday.bind(this);
    this.setFiveDay = this.setFiveDay.bind(this);
  }

  setToday() {
    this.setState(() => ({ showTodayWeather: true }));
  }

  setFiveDay() {
    this.setState(() => ({ showTodayWeather: false }));
  }

  render() {
    const { showTodayWeather } = this.state;
    const { zipcode } = this.props.match.params;

    return (
      <div style={{ color: 'rgb(255,255,255)' }}>
        <Navbar />

        <div className="nav">
          <button
            className={'button ' + (showTodayWeather ? 'active' : '')}
            onClick={this.setToday}
          >
            Today
          </button>
          <button
            className={'button ' + (showTodayWeather ? '' : 'active')}
            onClick={this.setFiveDay}
          >
            5-Day
          </button>
        </div>

        {showTodayWeather ? (
          <div className="weather-data">
            <Today data={zipcode} />
          </div>
        ) : (
          <div className="weather-data">
            <FiveDay data={zipcode} />
          </div>
        )}
      </div>
    );
  }
}

Forecast.defaultProps = {
  zipcode: '92092'
};

Forecast.propTypes = {
  zipcode: PropTypes.string
};

module.exports = Forecast;
