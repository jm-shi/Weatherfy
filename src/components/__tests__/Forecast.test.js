import React from 'react';
import Forecast from '../Forecast.js';
import { shallow } from 'enzyme';

test('should render Forecast correctly', () => {
  const match = { params: { zipcode: '92092' } };
  const wrapper = shallow(<Forecast match={match} />);
  expect(wrapper).toMatchSnapshot();
});
