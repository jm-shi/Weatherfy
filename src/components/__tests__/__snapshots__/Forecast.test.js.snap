// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Forecast correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Forecast
    match={
      Object {
        "params": Object {
          "zipcode": "92092",
        },
      }
    }
    zipcode="92092"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar />,
        <div
          className="nav"
        >
          <button
            className="button active"
            onClick={[Function]}
          >
            Today
          </button>
          <button
            className="button "
            onClick={[Function]}
          >
            5-Day
          </button>
        </div>,
        <div
          className="weather-data"
        >
          <Today
            data="92092"
          />
        </div>,
      ],
      "style": Object {
        "color": "rgb(255,255,255)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="button active"
              onClick={[Function]}
            >
              Today
            </button>,
            <button
              className="button "
              onClick={[Function]}
            >
              5-Day
            </button>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Today",
              "className": "button active",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Today",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5-Day",
              "className": "button ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "5-Day",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Today
            data="92092"
          />,
          "className": "weather-data",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": "92092",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar />,
          <div
            className="nav"
          >
            <button
              className="button active"
              onClick={[Function]}
            >
              Today
            </button>
            <button
              className="button "
              onClick={[Function]}
            >
              5-Day
            </button>
          </div>,
          <div
            className="weather-data"
          >
            <Today
              data="92092"
            />
          </div>,
        ],
        "style": Object {
          "color": "rgb(255,255,255)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="button active"
                onClick={[Function]}
              >
                Today
              </button>,
              <button
                className="button "
                onClick={[Function]}
              >
                5-Day
              </button>,
            ],
            "className": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Today",
                "className": "button active",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Today",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5-Day",
                "className": "button ",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "5-Day",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Today
              data="92092"
            />,
            "className": "weather-data",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": "92092",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
